### PR TITLE
Add telemetry for Jenkins uptime

### PR DESCRIPTION
**Pull request by daniel-beck**
@dwnusbaum and I noticed unexpected distribution of `correlator`s (see [JEP-214](https://github.com/jenkinsci/jep/tree/master/jep/214#specification)) in the submissions to our telemetry trials (my #8440 and #7470 and and Devin's #7342).

While most correlation IDs are submitted multiple times, up to once per day, there are outliers:

* A significant number of correlation IDs are only seen once, indicating very short-lived instances. JFR disables usage stats submission, so that's not it.
* A few correlation IDs are seen more than once per day during the collection period, sometimes many times per day. Our best guess so far is cloned Jenkins home directories, but some might also be frequently restarting instances.

In both cases (mostly the second though), collecting Jenkins uptime should be able to help understand what's going on.

This does not use `jenkins.model.Uptime` as data source, as that's not stable given system time changes (I filed [JENKINS-72157](https://issues.jenkins.io/browse/JENKINS-72157)).

### Testing done

```
def tel = ExtensionList.lookupSingleton(jenkins.telemetry.impl.Uptime)
def json = tel.createContent()
println ((json.get('now') - json.get('start')) / 1000 / 1000 / 1000)
```

Changed system time by 5 minutes, re-ran this, no significant jump.

Also locally changed the start date to Oct 10 and manually triggered submission (`ExtensionList.lookupSingleton(jenkins.telemetry.Telemetry.TelemetryReporter.class).run()`), then checked it in Uplink.

> Screenshot 2023-10-11 at 11 22 20

### Proposed changelog entries

- Add telemetry for Jenkins uptime.

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

File: core/src/main/resources/jenkins/telemetry/impl/Uptime/description.jelly
Comment (daniel-beck) [inline core/src/main/resources/jenkins/telemetry/impl/Uptime/description.jelly:10]: Slightly different wording than in the screenshot.

File: core/src/main/resources/jenkins/telemetry/impl/Uptime/description.jelly
Comment (dwnusbaum) [inline core/src/main/resources/jenkins/telemetry/impl/Uptime/description.jelly:3]: Why collect both endpoints instead of just the difference? I guess this makes it easier to tell when a single correlator is somehow being used by multiple instances because of overlapping but non-equal endpoints. Are there any other reasons?

**Review by dwnusbaum — APPROVED** [review]

File: core/src/main/resources/jenkins/telemetry/impl/Uptime/description.jelly
Comment (daniel-beck) [reply core/src/main/resources/jenkins/telemetry/impl/Uptime/description.jelly:3]: > I guess this makes it easier to tell when a single correlator is somehow being used by multiple instances because of overlapping but non-equal endpoints.

That's the idea. Multiple submissions will have the same start time if and only if Jenkins wasn't restarted in between.

**Review by NotMyFault — APPROVED** [review]
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Comment by MarkEWaite** [comment]
@daniel-beck this seems to me like it might be a good candidate for backporting to the 2.426.1 LTS.  Should it be an LTS candidate for 2.426.1 or would you prefer to wait for a later LTS release?

**Comment by daniel-beck** [comment]
@MarkEWaite Thanks for the reminder! I filed [JENKINS-72248](https://issues.jenkins.io/browse/JENKINS-72248). I'm happy to wait if it's already late for 2.426.1.

**Pull request merged**
